### ISSUE TITLE
All forms of UIA navigation appear to skip entries when going back and are unable to go forwards

### ISSUE THREAD

**Issue by ZoeyR** [microsoft/terminal#3976]
# Steps to reproduce

- Hook up the WPF control from the dev/zorio/wpf-uia-tree branch into a simple WPF application.
- Print text to the screen (e.g. "foo bar baz\r\nfoo2 bar2 baz2")
- Attempt to navigate said text with NVDA
# Expected behavior
Word, character and line navigation should work as expected.

# Actual behavior
All navigation forms appear to skip entries moving backwards and get stuck moving forwards, for example, I have observed the following:

- Reading the line "foo bar baz" from right to left as word units ends up reading out "baz, b, foo"
- Attempting to read the same line left to right reads out "foo" continuously.
- Reading the characters in that line will exhibit similar behaviour, unable to navigate rightwards and skipping entries when navigating leftwards.

**Comment by carlos-zamora** [comment]
@ZoeyR I'm having trouble navigating with NVDA (mostly because I'm not too familiar with how to use it well). I've tested this using Narrator and Inspect.exe and the text ranges seem to be accessible. Could you provide some additional guidance on how to navigate to the terminal's text field using NVDA?

I think I could probably figure out how to navigate within the text field using [this](https://dequeuniversity.com/screenreaders/nvda-keyboard-shortcuts). I'm just having trouble getting to it (for now).

**Comment by ZoeyR** [comment]
That reference doesn't have the full list of NVDA shortcut keys, what you want to do is use the desktop keyboard configuration, make sure that "caps lock" is set in preferences->keyboard->modifier keys. And then press Caps lock + numpad 5 to focus the current element.

**Comment by ZoeyR** [comment]
NVDA should also be the primary testing method, only 1% of computer users use narrator as their primary screen reader: https://webaim.org/projects/screenreadersurvey8/ I can help more if you're still having issues navigating. Also, you can't use the UWP application since the UIA tree has not yet been hooked up in that.

**Comment by carlos-zamora** [comment]
> NVDA should also be the primary testing method, only 1% of computer users use narrator as their primary screen reader: https://webaim.org/projects/screenreadersurvey8/ I can help more if you're still having issues navigating. Also, you can't use the UWP application since the UIA tree has not yet been hooked up in that.

I'll definitely use it more then. 😊 Just getting caught up in the learning curve haha.
